File: app/src/main/java/me/blog/korn123/easydiary/chart/MyAxisValueFormatter.java
package me.blog.korn123.easydiary.chart;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;

import java.text.DecimalFormat;

import me.blog.korn123.easydiary.R;

public class MyAxisValueFormatter implements IAxisValueFormatter
{

    private DecimalFormat mFormat;

    BarChartActivity barChartActivity;

    public MyAxisValueFormatter(BarChartActivity barChartActivity) {
        this.barChartActivity = barChartActivity;
        mFormat = new DecimalFormat("###,###,###,##0");
    }

    @Override
    public String getFormattedValue(float value, AxisBase axis) {
        return mFormat.format(value) + barChartActivity.getString(R.string.diary_count);
    }
}

File: app/src/main/java/me/blog/korn123/easydiary/chart/MyAxisValueFormatter.kt
package me.blog.korn123.easydiary.chart

import android.content.Context
import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.IAxisValueFormatter
import me.blog.korn123.easydiary.R
import java.text.DecimalFormat

class MyAxisValueFormatter(private var context: Context?) : IAxisValueFormatter {
    private val mFormat: DecimalFormat = DecimalFormat("###,###,###,##0")

    override fun getFormattedValue(value: Float, axis: AxisBase): String {
        return mFormat.format(value.toDouble()) + context?.getString(R.string.diary_count)
    }
}
